subt/depth2scan.py

```diff
@@ -99,16 +99,13 @@ def adjust_scan(scan, depth_scan, depth_params):
     DISAGREEMENT_MM_LIMIT = 600
     LIDAR_NO_MEASUREMENT = 0
     DEPTH_NO_MEASUREMENT = 0
-    # TODO: Get rid of unnecessary computation. We only want to check
-    #       disagreements at the beginning and end of the overlap.
-    disagreement = np.logical_and(
-        lidar_overlap != LIDAR_NO_MEASUREMENT,
-        np.logical_or(
-            depth_scan == DEPTH_NO_MEASUREMENT,
-            depth_scan - lidar_overlap > DISAGREEMENT_MM_LIMIT))
+    def disagreement(idx):
+        return lidar_overlap[idx] != LIDAR_NO_MEASUREMENT and (
+                depth_scan[idx] == DEPTH_NO_MEASUREMENT or
+                depth_scan[idx] - lidar_overlap[idx] > DISAGREEMENT_MM_LIMIT)
```

Comment:
I am not very fond of inner functions. Here it is probably ok but the inner functions in original local planner made it really difficult to read linearly and/or test the functions (preferably independently).